src/java/org/apache/cassandra/index/sai/disk/StorageAttachedIndexWriter.java

```diff
@@ -209,13 +204,11 @@ public void abort(Throwable accumulator)
      */
     public void abort(Throwable accumulator, boolean fromIndex)
     {
+        if (aborted) return;
+
         // Mark the write operation aborted, so we can short-circuit any further operations on the component writers.
         aborted = true;
         
-        // Make any indexes involved in this transaction non-queryable, as they will likely not match the backing table.
-        if (fromIndex)
-            indexes.forEach(StorageAttachedIndex::makeIndexNonQueryable);
-        
```

Comment:
The changes in `abort()` here are probably what require some real discussion. The position I'm going to try to defend here is that making the index non-queryable is sub-optimal in all cases if our goal is to keep indexes and backing tables consistent w/ one another. (I've fixed `SegmentFlushingFailureTester` to reflect this, but more tests are probably necessary...)

(I don't see any problem w/ continuing to not throw when the abort is upstream is generated from outside the index itself.)

@adelapena @mike-tr-adamson

Reply:
Haven't hit Memtable flush yet, but just added `IndexStreamingFailureTest`, which documents how SAI handles index build failures during streaming in the non-entire file case.

Reply:
I'm in favour of this change. I never felt that it should be down to the index to say whether or not it is queryable. Particularly at this point in the writer. Short of validating the index here we have no idea whether it is valid or not.